messaging/src/main/java/org/axonframework/commandhandling/gateway/NonTransientExceptionClassesPredicate.java

```diff
@@ -0,0 +1,75 @@
+package org.axonframework.commandhandling.gateway;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * A predicate for checking non-transiency of a failure comparing it against configured concrete classes.
+ * <p/>
+ * Comparison checks if any of configured classes is assignable from a tested failure instance.
+ * <p/>
+ * This predicate is intended to be used for configuring {@link RetryScheduler} implementations like in the following example:
+ * <pre>
+ * IntervalRetryScheduler myRetryScheduler = IntervalRetryScheduler
+ *      .builder()
+ *      .retryExecutor(new ScheduledThreadPoolExecutor(1))
+ *      .nonTransientFailurePredicate(
+ *          new NonTransientExceptionClassesPredicate(
+ *              AxonNonTransientException.class, NullPointerException.class, IllegalArgumentException.class,
+ *              IllegalStateException.class
+ *          )
+ *      )
+ *      .build();
+ * </pre>
+ *
+ * @author Damir Murat
+ * @since 4.6.0
+ */
+public class NonTransientExceptionClassesPredicate implements Predicate<Throwable> {
+  private final List<Class<? extends Throwable>> nonTransientFailures;
+
+  /**
+   * Initialize the predicate with class(es) that are considered as non-transient.
+   *
+   * @param nonTransientFailures vararg list of non-transient class(es)
+   */
+  @SafeVarargs
+  public NonTransientExceptionClassesPredicate(Class<? extends Throwable> ...nonTransientFailures) {
+    this.nonTransientFailures = Arrays.asList(nonTransientFailures);
+  }
+
+  @Override
+  public boolean test(Throwable failure) {
+    return isNonTransientFailure(failure);
+  }
+
+  /**
+   * Checks if provided failure is considered non-transient.
+   * <p/>
+   * This implementation checks if any configured exception class is assignable to the provided failure.
+   *
+   * @param failure The failure to check for non-transiency
+   * @return a boolean indicating if provided failure is non-transient ({@code true}) or not ({@code false})
+   */
+  protected boolean isNonTransientFailure(Throwable failure) {
```

Comment:
Nice! 
My argument for an interface was to have a `default` method implementation of this on the interface.
That it would be overwritten by this implementation is only reasonable, of course.